src/careamics/lightning/lightning_module.py

```diff
@@ -245,16 +245,20 @@ def __init__(self, algorithm_config: Union[FCNAlgorithmConfig, dict]) -> None:
         super().__init__()
         # if loading from a checkpoint, AlgorithmModel needs to be instantiated
         self.algorithm_config = (
-            FCNAlgorithmConfig(**algorithm_config)
+            VAEAlgorithmConfig(**algorithm_config)
             if isinstance(algorithm_config, dict)
             else algorithm_config
         )
 
         # create model and loss function
         self.model: nn.Module = model_factory(self.algorithm_config.model)
-        self.likelihood = likelihood_factory(self.algorithm_config.likelihood)
         self.noise_model = noise_model_factory(self.algorithm_config.noise_model)
-        self.loss_parameters = LVAELossParameters
+        self.noise_model_likelihood = likelihood_factory(
```

Comment:
@jdeschamps @veegalinova @melisande-c Take a look at this logic. Likelihoods aren't mutually exclusive and NMLL will take a noisemodel as input. 

Should NM be inside Pydantic then ?